src/App.tsx

```diff
@@ -1,14 +1,52 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { Filters } from './types/Filters';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+
+  const [isLoading, setIsLoading] = useState(true);
+
+  const [selectedTodoId, setSelectedTodoId] = useState<number | null>(null);
+
+  const [query, setQuery] = useState('');
+  const [status, setStatus] = useState<Filters>(Filters.All);
+
+  const onModalClose = () => {
+    setSelectedTodoId(null);
+  };
+
+  const selectedTodo = todos.find(todo => todo.id === selectedTodoId);
+
+  const filteredTodos = todos
+    .filter(todo => todo.title.toLowerCase().includes(query.toLowerCase()))
+    .filter(todo => {
+      if (status === Filters.All) {
+        return true;
+      }
+
+      return status === Filters.Completed ? todo.completed : !todo.completed;
+    });
+
+  useEffect(() => {
+    getTodos()
+      .then(data => {
+        setTodos(data);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
```

Comment:
Consider adding error handling for the `getTodos` promise. This will help manage any potential errors that occur during the data fetching process and provide feedback to the user if something goes wrong.